server/src/main/java/com/lunchtime/models/Restaurant.java

```diff
@@ -90,4 +86,35 @@ public Boolean isDeleted() {
     public void setDeleted(Boolean deleted) {
         isDeleted = deleted;
     }
+
+    public Restaurant() {    }
+
+    public Restaurant(Long id, String name,
+                      String email, String textAddress,
+                      String website, String description,
+                      String workingTime, Boolean isDeleted,
+                      Long menuId, Long ownerId,
+                      Integer tables, Float longitude,
+                      Float latitude, Instant createdAt,
+                      Long createdBy, Instant modifyAt,
+                      Long modifyBy) {
+
+        this.id = id;
+        this.name = name;
+        this.email = email;
+        this.textAddress = textAddress;
+        this.website = website;
+        this.description = description;
+        this.workingTime = workingTime;
+        this.isDeleted = isDeleted;
+        this.menuId = menuId;
+        this.ownerId = ownerId;
+        this.tables = tables;
+        this.longitude = longitude;
+        this.latitude = latitude;
+        this.createdAt = createdAt;
+        this.createdBy = createdBy;
+        this.modifyAt = modifyAt;
+        this.modifyBy = modifyBy;
+    }
```

Comment:
Why you added this constructor? You don't use this constructor.

Reply:
Because we need create object with initialization fields. For example user create restaurant and we use information to create this object. When developer use set/get he can forget one initialized field, but he have parameters constructor and he MUST enter all fields. Its more safer than use set/get.

Reply:
You can create a constructor with required parameters. In this case  param "id" will be created automatically, as well as the fields, that have default "null".

Reply:
For example in my cycle I have error when I not use setId
Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2020-03-18 22:26:35.065 ERROR 10998 --- [  restartedMain] o.s.boot.SpringApplication               : Application run failed

org.springframework.dao.InvalidDataAccessApiUsageException: detached entity passed to persist: com.lunchtime.models.Restaurant; nested exception is org.hibernate.PersistentObjectException: detached entity passed to persist: com.lunchtime.models.Restaurant
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.convertHibernateAccessException(HibernateJpaDialect.java:319) ~[spring-orm-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.springframework.orm.jpa.vendor.HibernateJpaDialect.translateExceptionIfPossible(HibernateJpaDialect.java:255) ~[spring-orm-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.translateExceptionIfPossible(AbstractEntityManagerFactoryBean.java:528) ~[spring-orm-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.springframework.dao.support.ChainedPersistenceExceptionTranslator.translateExceptionIfPossible(ChainedPersistenceExceptionTranslator.java:61) ~[spring-tx-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.springframework.dao.support.DataAccessUtils.translateIfNecessary(DataAccessUtils.java:242) ~[spring-tx-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:153) ~[spring-tx-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.springframework.data.jpa.repository.support.CrudMethodMetadataPostProcessor$CrudMethodMetadataPopulatingMethodInterceptor.invoke(CrudMethodMetadataPostProcessor.java:178) ~[spring-data-jpa-2.2.4.RELEASE.jar:2.2.4.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:95) ~[spring-aop-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186) ~[spring-aop-5.2.3.RELEASE.jar:5.2.3.RELEASE]
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:212) ~[spring-aop-5.2.3.RELEASE.jar:5.2.3.RELEASE]

Reply:
as I understand it, you don't use in this case constructor with parameters. You use default constructor and setMethods to create newEntity(). As result, you don't need constructor with parameters?!

Reply:
> MUST enter all fields

Not all. `website` can be blank, because restaurant doesn't need to have website. And restaurant can force clients booked tables in website, our project will be useless in this case

Reply:
I use constructor in seed , and entered all fields. I create full restaurant.

Reply:
You can use **[@NoArgsConstructor](https://projectlombok.org/api/lombok/NoArgsConstructor.html)** or **[@AllArgsConstructor](https://projectlombok.org/api/lombok/AllArgsConstructor.html)** here. Here is great **[article](http://stevenmwesigwa.com/tutorials/project-lombok/8/how-to-use-allargsconstructor-annotation-with-project-lombok-in-java-applicat)** about `@AllArgsConstructor` usage practice.

Reply:
I didn't know this magic) This stylistic code I apply in future. At now this code work. Thanks.

Reply:
To be honest, I can't put up with this code. Moreover, annotation power it will be a good example for everybody looks this code. Come on bro, try it :)